Update to angular 15

NicMcPhee opened the pull request:
This basically copies over the latest files from [the Stackblitz project](https://angular.io/generated/live-examples/getting-started-v0/stackblitz.html) and merges those changes into our project.

Other than jumping up to V15 of Angular, most of this is pretty minor. A few of the highlights are:

- They seemed to have removed the environment files altogether (which we were never using)
- They've replaced the use of `polyfills.ts` with `zone.js`
- They've removed some testing bits; I suspect they gave up on adding tests to this, at least for now? 
- They changed the name of the project to `angular.io-example` (no particular idea why)

In making these changes I appear to have removed all support for `ng lint` by mistake. Ooops. Happily, running `ng add @angular-eslint/schematics` totally fixed things locally. Unfortunately our GitHub Actions check still failed.

NicMcPhee committed:
Remove files that aren't in the tutorial anymore

They've removed some environment files (which we never used), `polyfills.ts` (which I assume we don't need for some reason?), and `test.ts`, which makes sense since they never had any tests that ran or passed.

NicMcPhee committed:
Update `angular.json`

Various changes they've made in the latest version of the tutorial:

- They changed the name of the project to `angular.io-example`
- They dropped the `"schematics" section
- They replaced `polyfills.ts` with `zone.js`
- They dropped the use of environments
- They removed most of the test bits (since there are not actually any tests)

NicMcPhee committed:
Update all the Angular packages

Their `package.json` has lots of RC-1 references. I suspect we can update those to "real" v15 packages once we have this working.

NicMcPhee committed:
Update all the other files

A number of minor changes. Most of the other changes were the addition of the Google copyright notices, which I removed.

NicMcPhee committed:
Convert from v15-rc.1 to v15

This removes all the references to `rc-1` in the Angular versions.

NicMcPhee committed:
Restore `ng lint`

I appear to have removed all the `ng lint` support. I just did `npm add @angular-eslint/schematics` and that seemed to have fixed everything.

NicMcPhee committed:
Merge branch 'main' into update-to-angular-15

jpwalbran reviewed (approved):
Looks good!

kklamberty reviewed (commented):
I will run this in the lab tomorrow before approving the changes, but everything looks good to me.

kklamberty committed:
Add instructions for doing this work in our lab

Since we don't have a current enough version of Node running (and there isn't time to make sure we do for this lab), I've added instructions for the students to set up nvm and manage Node versions to the extent that is needed for this lab exercise.

kklamberty committed:
Make setting up nvm and node a minor step

Since this is a minor part of setting things up rather than a major step in the lab, I reorganized it slightly.

kklamberty commented on src/app/app.module.ts:
```diff
@@ -13,7 +13,7 @@ import { ProductListComponent } from './product-list/product-list.component';
     ReactiveFormsModule,
     RouterModule.forRoot([
       { path: '', component: ProductListComponent },
-    ], { useHash: true })
```
Apparently, we need this for GH pages to work

kklamberty commented on src/index.html:
```diff
@@ -3,6 +3,7 @@
   <head>
     <meta charset="utf-8" />
     <title>Angular Getting Started</title>
+    <base href="/" />
```
And, we need to remove this for GH Pages to work.

kklamberty reviewed (commented):
We broke GH Pages with this commit, but somehow we didn't notice. @floogulinc found what to fix, and I've marked them in this review.
